Balloonduino.cpp

```diff
@@ -8,47 +8,171 @@
+        initialized = false;
+    }
+    else
+    {
+        debug_serial.println("DS1308 RTC initialized.");
+    }
+
+    if (!rtc.isrunning())
+    {
+        debug_serial.println("WARNING: RTC is NOT running!");
+        // following line sets the RTC to the date & time this sketch was compiled
+        rtc.adjust(DateTime(F(__DATE__), F(__TIME__)));
+        // This line sets the RTC with an explicit date & time, for example to set
+        // January 21, 2014 at 3am you would call:
+        // rtc.adjust(DateTime(2014, 1, 21, 3, 0, 0));
+    }
+
+    //// SoftRTC (for subsecond precision)
+    SoftRTC.begin(rtc.now());    // Initialize SoftRTC to the current time
+    start_millis = millis();    // get the current millisecond count
+#endif
+
+    //// Init BME environmental sensor
+#ifndef BALLONDUINO_NO_BME
+    if (!bme.begin(0x76))
+    {
+        debug_serial.println("WARNING: BME280 initialization failure.");
+        initialized = false;
+    }
+    else
+    {
+        debug_serial.println("BME280 initialized.");
+    }
+#endif
+
+    //// Init ADS power monitor
+#ifndef BALLONDUINO_NO_ADS
+    ads.begin();
+    ads.setGain(GAIN_ONE);
+    debug_serial.println("ADS1015 initialized.");
+#endif
+
+    //// Init SSC pressure sensor
+    //  set min / max reading and pressure, see datasheet for the values for your
+    //  sensor
+#ifndef BALLONDUINO_NO_SSC
+    ssc.setMinRaw(0);
+    ssc.setMaxRaw(16383);
+    ssc.setMinPressure(0.0);
+    ssc.setMaxPressure(30);
+
+    //  start the sensor
+    debug_serial.print("SSC start: ");
+    debug_serial.println(ssc.start());
+#endif
+
+    //// Init XBee radio
+    debug_serial.println("Beginning xbee init");
 
-    Serial.print("[");
-    if (hours < 10)
+    int xbeeStatus = InitXBee(XBEE_ADDR, XBEE_PAN_ID, xbee_serial, false);
+    if (!xbeeStatus)
+    {
+        debug_serial.println("XBee initialized.");
+    }
+    else
     {
-        Serial.print("0");
+        debug_serial.print("ERROR: XBee initialization error with code: ");
+        debug_serial.println(xbeeStatus);
+        initialized = false;
     }
-    Serial.print(hours);
-    Serial.print(":");
-    if (minutes < 10)
+
+    //// Init MicroSD storage
+    SPI.begin();
+    pinMode(53, OUTPUT);
+    if (!SD.begin(53))
     {
-        Serial.print("0");
+        debug_serial.println("SD Card initialization failure.");
+        initialized = false;
     }
-    Serial.print(minutes);
-    Serial.print(":");
-    if (seconds < 10)
+    else
     {
-        Serial.print("0");
+        debug_serial.println("SD Card initialized.");
     }
-    Serial.print(seconds);
-    Serial.print("] ");
+
+    // MicroSD
+    // appends to current file
+    // NOTE: Filenames must be shorter than 8 characters
+    File IMULogFile = SD.open("IMU_LOG.txt", FILE_WRITE);
+    File PWRLogFile = SD.open("PWR_LOG.txt", FILE_WRITE);
+    File ENVLogFile = SD.open("ENV_LOG.txt", FILE_WRITE);
+
+    return initialized;
```

Comment:
this function returns an int, but the return value is being set to a boolean. Should be made consistent

Reply:
Isn't bool cast to 0 or 1 implicitly? I figured it would be simpler for readability to use a bool throughout the function, but all the begin()s I've ever seen return an int